titan/interactions/injection.py

```diff
@@ -25,6 +25,11 @@ def interact(model: "model.HIVModel", rel: "agent.Relationship") -> bool:
         if model.time < model.params.hiv.start_time:
             return False
 
+        assert (
+            "injection"
+            in rel.agent1.location.params.classes.bond_types[rel.bond_type].acts_allowed
```

Comment:
this should probably point to `model` params instead of agent.  Is it really needed?

Reply:
It *should* only get here if injection is in the acts_allowed, I guess.

Reply:
I think the dispatch to these methods is based on the bond type def, so this assertion shouldn't be needed (probably?)